pyro/nn/auto_reg_nn.py

```diff
@@ -83,8 +83,11 @@ def __init__(self, input_dim, hidden_dim, output_dim_multiplier=1,
         self.mask2 = torch.zeros(input_dim * self.output_dim_multiplier, hidden_dim)
 
         for k in range(hidden_dim):
-            # fill in mask1
-            m_k = self.mask_encoding[k].item()
+            if self.mask_encoding.dim():
```

Comment:
why is this needed?

Reply:
you cant index a scalar.

Reply:
i'd remove this. cuz of an IAF is 1D it'll just be an affine transformation of the base distribution. raise an error if user wants 1d autoiaf?

Reply:
we can catch this for auto iaf, but like you said a 1d iaf is a valid thing, and even if it werent that error should be caught, not throw a pytorch tensor error.